### PR TITLE
feat(ci): Add release version guards

### PR DESCRIPTION
**Pull request by RPate97**
## Purpose
Adds release guard rails, so we are forced to always make a `plugins` package release at the same time that we make a

### CODE REVIEW

File: test/changesets.spec.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    const pluginsRelease = changesets.releases.find(
+      (release) => release.name === '@sphinx-labs/plugins'
+    )
+
+    if (coreRelease && pluginsRelease) {
+      expect(mapReleaseTypeToIndex(coreRelease.type)).to.be.greaterThanOrEqual(
+        mapReleaseTypeToIndex(pluginsRelease.type)
+      )
+    }
+  })
```
[the remark this answers is not in the document]
Comment (sam-goldman) [reply test/changesets.spec.ts:126]: And I guess the new test case that compares plugins and contracts too

**Review by sam-goldman — COMMENTED** [review]
Looks good, just seems like there may be an edge case to cover
